### ISSUE TITLE
k8s application status updated only when update-status hook is triggered

### ISSUE THREAD

**Issue by hemanthnakkina** [canonical/k8s-operator#61]
### Bug Description

Deployed k8s using steps mentioned in https://github.com/canonical/k8s-operator/tree/main/charms/worker#deploy

After couple of minutes, the command `sudo k8s status` shows `ready` but `juju status` shows as `waiting`.
juju status of application k8s gets updated to active only after 5 minutes when the update-status hook is triggered.

Is it possible to update the k8s application status in juju to active as and when sudo k8s status is ready?

### To Reproduce

Follow steps in https://github.com/canonical/k8s-operator/tree/main/charms/worker#deploy to deploy k8s

Observe the following to see when the status became ready/active
juju debug-log -i unit-k8s-0
sudo k8s status

### Environment

juju 3.4/stable + k8s charm latest/edge rev 27 

### Relevant log output

```shell
.
```

### Additional context

sunbeam deploys k8s application using terraform and then waits for juju status of application to be active. And now it takes 5 minutes for sunbeam to consider k8s as active.

**Comment by addyess** [comment]
there's no trigger from the k8s snap on the machine which can in real-time trigger a juju agent hook.  The `update-status` hook is fired on an interval for just this reason. 

sunbeam's terraform can adjust the interval at which update-status events are fired until the application is active/idle

```
juju model-config update-status-hook-interval=60s
```

**Comment by addyess** [comment]
Duplicate of https://github.com/canonical/k8s-operator/issues/90
